tests/scripts/components-build-system.sh

```diff
@@ -108,10 +108,15 @@ component_test_cmake_as_package () {
     make neat
 
     msg "build: cmake 'as-package' build"
+    root_dir="$(pwd)"
     cd programs/test/cmake_package
+    build_variant_dir="$(pwd)"
     cmake .
     make
     ./cmake_package
+    if [[ "$OSTYPE" == linux* ]]; then
+        PKG_CONFIG_PATH="${build_variant_dir}/mbedtls/pkgconfig" ${root_dir}/tests/scripts/pkgconfig.sh
+    fi
```

Comment:
Any reason why this is put in `component_test_cmake_as_package()` here for the `development` PR but [in `component_test_cmake_as_package_install()` on the 3.6 backport](https://github.com/Mbed-TLS/mbedtls/pull/9174/files#diff-155c17ab3dd1c2acbcb5d65c25288695df8e59e9892e5d15aa0c12c7e8e28af3R133)?

Reply:
Incorrect resolution to the conflict on the rebase, and the incorrect logic also followed to the 2.28 backport, but has been fixed there.